pkg/machine/hyperv/stubber.go

```diff
@@ -55,12 +56,18 @@ func (h HyperVStubber) CreateVM(opts define.CreateVMOpts, mc *vmconfigs.MachineC
 		Memory:   uint64(mc.Resources.Memory),
 	}
 
-	networkHVSock, err := vsock.NewHVSockRegistryEntry(mc.Name, vsock.Network)
-	if err != nil {
-		return err
-	}
+	if opts.UserModeNetworking {
```

Comment:
This is code-wise correct but this could change the current behavior in podman. The default value for UserModeNetworking is false, so if we merge this upstream, it would change the behavior of what users are facing right now.

Reply:
Yep, would be better not to change the default behaviour. The --user-mode-networking is handled as a tri-state in `InitOpts`, so we can differentiate between enabled/disabled/use default.
`UserModeNetworkEnabled` is only relevant for WSL at the moment, otherwise it’s always true, dunno if this interface method could be reworked. 
Imo the machine config could indicate if usermode networking is enabled or not, the stubber would return an error if it did not manage to set the network mode the config asked for.
Might be interesting to know which networking modes the stubber supports, and which one it wants to use by default.

Reply:
An alternative could be to hardcode the system networking by default in macadam for hyperv, but that’s ugly too